src/main/java/seedu/address/model/lesson/TimeRange.java

```diff
@@ -82,12 +87,18 @@ private boolean isEndClashing(Time time) {
      * @param other The TimeRange to be tested.
      */
     public boolean isClashing(TimeRange other) {
-        return isStartClashing(other.start) || isEndClashing(other.end);
+        if (start.compareTo(other.start) >= 0 && start.compareTo(other.end) <= 0) {
+            return true; // Start time is within the other time range
+        }
+        if (end.compareTo(other.start) >= 0 && end.compareTo(other.end) <= 0) {
+            return true; // End time is within the other time range
+        }
+        return false;
```

Comment:
![image](https://user-images.githubusercontent.com/62977276/135481677-30d411f2-ae34-4853-a585-3f4de8d0ce4b.png)
The new logic does not add a lesson from 1200-XXXX if there is an existing lesson from XXXX-1200.

Steps to replicate:

1. have a lesson that starts at any time before 1200 (after 0800) and ends at 1200.
2. ladd a lesson that starts at 1200 and end any time after 1200 (before 2200)